asf_heat_pump_suitability/getters/get_dno_datasets.py

```diff
@@ -146,14 +156,13 @@ def generate_npg_gdf() -> gpd.GeoDataFrame:
         (npg_demand["bulk_supply_point_or_primary"] == "Primary")
         & (npg_demand["scenario_name"] == "NPg Best View")
     ]
-    npg_demand["geo_point_2d"] = npg_demand["geo_point_2d"].apply(parse_binary_geometry)
     npg_demand["geo_shape"] = npg_demand["geo_shape"].apply(parse_binary_geometry)
-    npg_shape = npg_demand[["geo_point_2d", "geo_shape"]].drop_duplicates()
+    npg_shape = npg_demand["geo_shape"].drop_duplicates()
 
     # Create GeoDataFrames and perform spatial join
     npg_gdf = gpd.GeoDataFrame(npg, geometry="location", crs=CRS)
     npg_shape_gdf = gpd.GeoDataFrame(npg_shape, geometry="geo_shape", crs=CRS)
-    npg_df = gpd.sjoin(npg_shape_gdf, npg_gdf, how="left", predicate="intersects")
+    npg_df = gpd.sjoin(npg_shape_gdf, npg_gdf, how="left", predicate="within")
```

Comment:
```suggestion
    npg_df = gpd.sjoin(npg_shape_gdf, npg_gdf, how="left", predicate="contains")
```
There is a bug here which can be fixed by changing the predicate I think. Currently, the function returns a df of len 0. I think the current predicate is trying to polygons from `npg_shape_gdf` located _within_ point geoms in `npg_gdf` which is not matching anything.

Reply:
This also makes me think we could add an AssertionError in `grid_capacity.generate_substations_gdf` (or something similar) to raise an error if any of the expected DNOs are missing from the `operator` column in the final `gdf`. Otherwise this could be a silent fail.